internal/core/window.rs

```diff
@@ -321,6 +378,9 @@ impl WindowInner {
     pub fn process_mouse_input(&self, mut event: MouseEvent) {
         crate::animations::update_animations();
 
+        // handle multiple press release
+        event = self.click_state.check_repeat(event);
```

Comment:
Instead of passing through the event and modifying only the click-count, perhaps it would be easier to write

`event.click_count = self.click_state.compute_click_count(&event);`

?

That it's clearer IMO that all this does is compute a number (and happens to update interior state), but doesn't change anything else of the event.

Reply:
@tronical because  `MouseEvent` is an enum and not all variants of it have a count.